### PR TITLE
Golang Template Fixes - Lambda 

### PR DESCRIPTION
**Pull request by Johnbush**
### Requirements

- Fixing Templates issues for golang runtime.

### Description of the Change

- Added error Handler wrapper 

- Implemented Logging and Logging Levels

- Attached Config files, also added loading config files based on environment.

### CODE REVIEW

File: templates/lambda-template-go/config.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	if value, ok := event["stage"]; ok {
+		//get stage value from payload
+		stage = value.(string)
+	} else {
+		// get Function name from Environment Variables 
+		FunctionName = os.Getenv("AWS_LAMBDA_FUNCTION_NAME")
+		fnName := string(FunctionName [strings.LastIndex(FunctionName , "-") + 1 :len(FunctionName )])
+		if (strings.HasPrefix(fnName, "dev")){
+			stage = "dev"
+		}else if (strings.HasPrefix(fnName, "stg")){
```
Comment (svidhani) [inline templates/lambda-template-go/config.go]: indentation

File: templates/lambda-template-go/error-handler.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// NotFound Error
+type NotFoundError struct {
+	message string
+}
+func NewNotFoundError(message string) *NotFoundError {
+    return &NotFoundError{
+        message: message,
+    }
+}
+func (e *NotFoundError) Error() string {
```
Comment (svidhani) [inline templates/lambda-template-go/error-handler.go]: try to give a new line after every function

File: templates/lambda-template-go/logger.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+import (
+	"log"
+	"os"
+	"fmt"
+	"time"
+)
+
+var loglevels = map[string]int{
+	"Error": 4,
```
Comment (svidhani) [inline templates/lambda-template-go/logger.go]: indentation here and other places

File: templates/lambda-template-go/logger.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+func logWithFormater(logLevel string , message string){
+	log.SetFlags(0)
+	// New Instance of Writer Method
+	logger := new(logWriter)
+	// set the Loglevel to be append dynamically
+	logger.init(logLevel)
+	// Set the Ouptut Formatter
+	log.SetOutput(logger)
+	log.Println(message)
+	//logger.Println(message)
```
Comment (svidhani) [inline templates/lambda-template-go/logger.go]: remove

**Comment by svidhani** [comment]
Seems like we can't have the folder structure like in nodejs or java

File: templates/lambda-template-go/logger.go (hunk 4 above)
Comment (Johnbush) [reply templates/lambda-template-go/logger.go]: Removed

File: templates/lambda-template-go/config.go (hunk 1 above)
Comment (Johnbush) [reply templates/lambda-template-go/config.go]: fixed

**Comment by svidhani** [comment]
the folder name is still configs?

**Comment by Johnbush** [comment]
No it's already updated to config.

**Pull request merged**